### PR TITLE
imp: disallow submessage execution

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

Item 4.4 of Confio's report of 08-wasm audit.

This check prevents Wasm light client contracts to execute sub messages.

closes: #XXXX

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/light-clients/08-wasm/types/client_state.go [hunk 1]
```diff
@@ -261,5 +261,8 @@ func call[T ContractResult](ctx sdk.Context, clientStore sdk.KVStore, cs *Client
 	if !output.Validate() {
 		return output, errorsmod.Wrapf(errors.New(output.Error()), "error occurred while calling contract with code ID %s", hex.EncodeToString(cs.CodeId))
 	}
+	if len(out.Messages) > 0 {
+		return output, errorsmod.Wrapf(ErrWasmSubMessagesNotAllowed, "code hash (%s)", hex.EncodeToString(cs.CodeId))
```
Comment (crodriguezvega) [inline modules/light-clients/08-wasm/types/client_state.go:265]: I mention code hash in the message, assuming that [this PR](https://github.com/cosmos/ibc-go/pull/4091) will eventually get merged.

**Review by DimitrisJim — APPROVED** [review]
looks good to go after #4091

File: modules/light-clients/08-wasm/types/client_state.go (hunk 1 above)
Comment (damiannolan) [reply modules/light-clients/08-wasm/types/client_state.go:265]: The var naming of `out` doesn't really feel intuitive to me, but we can rename it to whatever later, also maybe we could add something like this in a future improvement - `handleContractResponse()` [ref](https://github.com/CosmWasm/wasmd/blob/main/x/wasm/keeper/keeper.go#L983)

**Review by damiannolan — APPROVED** [review]
LGTM

**Pull request merged**